playbooks/roles/openstack-remove-compute/tasks/main.yml

```diff
@@ -4,46 +4,44 @@
      set -o pipefail
      openstack compute service list | grep {{ compute_node_name }} | awk '{print $2}'
   register: compute_id
-  delegate_to: localhost
+  delegate_to: "{{ groups['soc-deployer'][0] }}"
   failed_when: false
   changed_when: false
 
 - name: Validate compute node id.
   debug:
     msg: "Compute node {{ compute_node_name }} does not exists in openstack."
-  delegate_to: localhost
   when: compute_id.stdout | length | int == 0
 
 - name: Get VM list from compute node.
   command: "openstack server list --host {{ compute_node_name }}"
   register: vm_exist
-  delegate_to: localhost
+  delegate_to: "{{ groups['soc-deployer'][0] }}"
   failed_when: false
   changed_when: false
 
 - name: Validate VM(s) exists on compute node.
   fail:
     msg: "VM(s) exists on compute node {{ compute_node_name }}."
-  delegate_to: localhost
```

Comment:
Everywhere else we change the delegate_to, but here we just delete it. Why is this one different?

Reply:
There are two places where I removed it instead of changing the target: the debug module above and here at the fail module. Both of these module are kinda special in that they don't really execute anything on the target host, so to me it doesn't make sense to specify a delegate host for those modules.

Reply:
Thank you for the explanation.